### ISSUE TITLE
G4 doorbell no doorbell notifications on > v6.9.2 

### ISSUE THREAD

**Issue by bluewalk** [hjdhjd/homebridge-unifi-protect#961]
**Describe The Problem:**
On versions higher than v6.9.2 the doorbell notifications and doorbell chime options are gone in the Home app. No notifications are sent to homepods nor to apple tv's.

**To Reproduce:**
Update to a version higher than v6.9.2 

**Logs:**
Logs are actually the same between the two versions. I haven't checked debuglogs yet though.

**Homebridge Configuration:**

```json
{
    "bridge": {
        "name": "Homebridge XXXX",
        "username": "0E:31:6C:XX:XX:XX",
        "port": 53063,
        "pin": "999-99-999",
        "advertiser": "bonjour-hap",
        "bind": [
            "ens160"
        ]
    },
    "accessories": [],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "auth": "form",
            "theme": "auto",
            "tempUnits": "c",
            "lang": "en",
            "platform": "config"
        },
        {
            "name": "Living Room",
            "hubIP": "192.168.1.202",
            "DELAY_BEFORE_RETRY_AFTER_NETWORK_LOSS": 60000,
            "HUB_CONNECT_TIMEOUT": 10000,
            "HUB_SEND_TIMEOUT": 30000,
            "publishGeneralMuteSwitch": true,
            "linkVolumeControlToTV": true,
            "platform": "HarmonyHubWebSocket"
        },
        {
            "email": "",
            "password": "",
            "platform": "NeatoVacuumRobot",
            "hidden": [
                "schedule",
                "spot"
            ]
        },
        {
            "controllers": [
                {
                    "address": "172.16.1.254",
                    "mqttUrl": "mqtt://mqttserver",
                    "password": "password",
                    "username": "username"
                }
            ],
            "name": "UniFi Protect",
            "platform": "UniFi Protect"
        }
    ],
    "disabledPlugins": []
}
```

**Screenshots:**
v6.9.2 - G4 doorbell options:
![IMG_2489](https://github.com/hjdhjd/homebridge-unifi-protect/assets/6522477/799e893d-7e77-4f18-b68f-79009ddeacda)

after v6.9.2:
![IMG_2487](https://github.com/hjdhjd/homebridge-unifi-protect/assets/6522477/8478de92-dc26-45b5-b89a-a165dafbc043)

**Environment:**

* **Homebridge Version:** v1.6.1 
* **Node Version:** v18.13.0
* **Homebridge-UniFi-Protect Plugin Version:** > v6.9.2 
* **UniFi Protect Version:** 2.6.17
* **UniFi Protect Controller Type:** UDMP
* **UniFi Protect Controller Firmware Version:** v3.0.20
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:** 16.5.1
* **Operating System and OS Version:** Ubuntu + docker

**Comment by bluewalk** [comment]
Updated to:
**UniFi Protect Version:** 2.8.35
**UniFi Protect Controller Firmware Version:** v3.1.14

Problem still persists

**Comment by hjdhjd** [comment]
Appreciate the report, but it's missing a crucial piece of information. 😄 

What cameras and doorbells do you have connected to the Protect controller, and which one(s) are having an issue?

Thanks - I know what's going on here, in general terms...but let's figure out what's going on in your configuration and we'll get this working.

**Comment by bluewalk** [comment]
Hi @hjdhjd , my apologies, I forgot that information.
Currently my Protect has 4 camera's running:
* 3x G3 Bullet (firmware 4.64.150)
* 1x G4 Doorbell (non pro, the white one) (firmware 4.64.150)

The G4 doorbell shows up in HomeKit as a doorbell with v6.9.2 and I can choose to enable the doorbell notification and chime. Once I update the plugin to the latest version (or any version above v.6.9.2) these options disappear from the Home app (see screenshots). When pressing the doorbell no notifications are sent nor do the HomePods chime.

**Comment by hjdhjd** [comment]
Let me recreate the same environment on my end and see if I can duplicate the outcome. I think I know what's going on here. In the meantime, feel free to use 6.9.2 until then...won't be until later this week, but really appreciate your thorough testing between versions to this point.

**Comment by bluewalk** [comment]
No worries, if you need me to test anything just let me know 😀

**Comment by sebwar2004** [comment]
Hi,
first: Thanks for your great work!

Unfortunately I have to add that I have the same issues (I'm on 6.11.0) with missing Doorbell options and also missing notifications, for example on the AppleTV.

Devices are:
- G4 Doorbell
- G4 Instant
- AI 360
with UDM SE.

**Comment by hjdhjd** [comment]
@sebwar2004 Thanks for your report -  I don’t need more information or additional data points on this one right now. Just stay tuned and monitor this issue. Thanks!

**Comment by sebwar2004** [comment]
Thanks for the fast reply!

I have to say: today evening it works again after a week. So I'm not quite sure what was causing this in my setup...

**Comment by bluewalk** [comment]
@hjdhjd Today I saw 6.11.1 got released so why not give that one a shot. And presto, the doorbell works again!

**Comment by hjdhjd** [comment]
> @hjdhjd Today I saw 6.11.1 got released so why not give that one a shot. And presto, the doorbell works again! This doesn't really make sense looking at the changes though.
> 
> [edit] One other thing, I did recently upgrade to the official release of v3.1.15, maybe that fixed some issues?

So...I actually was able to recreate this issue. You are correct that the bug / issue here was actually in Protect firmware - namely Ubiquiti apparently forgot to indicate that the G4 Doorbell was in fact a doorbell in recent firmware releases. That's fixed in the 3.1 UniFi OS releases that are now GA. I've also confirmed that this is no longer an issue so long as you're on the latest GA UniFi OS and Protect controller firmwares - which should upgrade the firmware on the G4 Doorbell to fix things.

The bugfixes in yesterday's HBUP update had nothing to do with this issue. 😄

TL;DR: bug was in Ubiquiti/Protect firmware and is fixed with the latest GA Protect controller and UniFi OS firmwares.

Thanks for a cool report!

**Comment by bluewalk** [comment]
Thank you so much for looking into this, this is still my favorite Homebridge plugin 😁

**Comment by hjdhjd** [comment]
Glad you do - pay it forward by sharing the awesomeness with others and spreading the word. 😄
